### ISSUE TITLE
Как ограничить общее потребление памяти процессом ClickHouse

### ISSUE THREAD

**Issue by yurchenkosv** [yandex/ClickHouse#1632]
Добрый день! Используем ClickHouse для хранения логов кластера, ClickHouse запущен как Pod кластера Kubernetes и нам не нужно, чтобы он использовал много памяти ноды. Проблема в том, что после запуска  ClickHouse начинает занимать память и не отдает ее обратно, что в конечном итоге приводит к тому, что нода падает по OutOfMemory. Не хотелось бы завершать процесс Clickhouse при достижении некоторого лимита памяти ноды, это будет очень грубое решение. 
Отсюда вопрос: каким образом можно ограничить общее использование памяти сервисом ClickHouse?

**Comment by kshvakov** [comment]
Совсем не отдает обратно ? Так быть не должно, но, если у вас 54310, то там есть проблема с нереплицируемыми таблицами MergeTree, там как раз есть утечка памяти

**Comment by yurchenkosv** [comment]
Огромное спасибо за ответ! Действительно, версия 54310. Будем пробовать обновиться до v1.1.54318-stable.

**Comment by kshvakov** [comment]
@yurchenkosv Помогло?

**Comment by yurchenkosv** [comment]
@kshvakov Да, спасибо, после переезда на 1.1.54318-stable, проблема решилась, clickhouse сейчас использует ~600мб и замечательно отдает память.
